tests/Collection/FunctionalTestCase.php

```diff
@@ -19,15 +19,6 @@ public function setUp(): void
 
         $this->collection = new Collection($this->manager, $this->getDatabaseName(), $this->getCollectionName());
 
-        $this->dropCollection();
-    }
-
-    public function tearDown(): void
-    {
-        if (! $this->hasFailed()) {
-            $this->dropCollection();
-        }
-
-        parent::tearDown();
+        $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
```

Comment:
```suggestion
        $this->collection = $this->dropCollection($this->getDatabaseName(), $this->getCollectionName());
```

Along with removing the line above (can't suggest since it's outside of the PR diff).

I saw you do this in ChangeStreamIteratorTest and elsewhere.